### PR TITLE
Add support new native majors

### PR DESCRIPTION
**Pull request by tonidero**
This bumps PHC to 13.0.0 which includes breaking changes in the native SDKs: 
- iOS v5: https://github.com/RevenueCat/purchases-ios/releases/tag/5.0.0
- Android v8: https://github.com/RevenueCat/purchases-android/releases/tag/8.0.0

### CODE REVIEW

**Comment by tonidero** [comment]
Want to test this better but this should be ready for review

**Review by vegaro — APPROVED** [review]
Didn't see anything to call out

File: RevenueCat/Scripts/Purchases.cs [hunk 1]
```diff
@@ -143,11 +134,11 @@ private void Configure(string newUserId)
         var dangerousSettings = new DangerousSettings(autoSyncPurchases);
         var builder = PurchasesConfiguration.Builder.Init(apiKey)
             .SetAppUserId(newUserId)
-            .SetObserverMode(observerMode)
+            .SetPurchasesAreCompletedBy(purchasesAreCompletedBy, storeKitVersion)
```
Comment (fire-at-will) [inline RevenueCat/Scripts/Purchases.cs:137]: 🤩🤩

**Review by fire-at-will — CHANGES_REQUESTED** [review]
With the note that I'm _not_ at all familiar with Unity, all of the changes that you have here look good to me!

There is one function from iOS native 5.X that we should also bring in to be safe, [recordPurchase](https://github.com/RevenueCat/purchases-hybrid-common/blob/2a04578efaa6ff668c017b69f0224538c8118db4/ios/PurchasesHybridCommon/PurchasesHybridCommon/CommonFunctionality.swift#L507). Let me know if you have any questions about the function and I'll be happy to help!

**Comment by tonidero** [comment]
Ah yes thanks for the reminder! Will add that now

**Review by fire-at-will — APPROVED** [review]

**Comment by fire-at-will** [comment]
The migration guide for Unity should be pretty similar to the react native one, just with Unity code samples instead of RN: https://github.com/RevenueCat/react-native-purchases/blob/main/migrations/v8-MIGRATION.md

File: RevenueCat/Scripts/Purchases.cs [hunk 2]
```diff
@@ -140,14 +131,20 @@ private void Configure(string newUserId)
                  || IsAndroidEmulator())
             apiKey = useAmazon ? revenueCatAPIKeyAmazon : revenueCatAPIKeyGoogle;
 
+        if (purchasesAreCompletedBy == PurchasesAreCompletedBy.MyApp && storeKitVersion == StoreKitVersion.Default)
+        {
+            Debug.Log("You must set a StoreKit version if you are setting PurchasesAreCompletedBy to MyApp. For Android, it doesn't matter which");
```
Comment (tonidero) [inline RevenueCat/Scripts/Purchases.cs:136]: A couple things to note here, when configuring through the Editor:
- There is no breaking change even after removing the old observer mode. So if someone was setting observer mode in the editor before, there won't be any compilation issue when they migrate to the new version.
- There is also no compile-time check that makes it so you can't set "PurchasesAreCompletedBy.MyApp" without setting storeKitVersion as well.
For these 2 reasons, we're adding this runtime check which would make it so we don't configure the SDK in this situation and log this message.

**Pull request merged**